tosfs/core.py

```diff
@@ -203,6 +203,59 @@ def info(
 
         return self._try_dir_info(bucket, key, path, fullpath)
 
+    def rmdir(self, path: str) -> None:
+        """Remove a directory if it is empty.
+
+        Parameters
+        ----------
+        path : str
+            The path of the directory to remove. The path should be in the format
+            `tos://bucket_name/directory_path`.
+
+        Raises
+        ------
+        FileNotFoundError
```

Comment:
So in fsspec,  we will raise `FileNotFoundError` even if it is a directory, right ?

Reply:
In fsspec, this method does not provide a template implementation and has too many constraint requirements. In the built-in exception definitions of the Python language, there is no strict distinction between dir and file, and there are also examples of throwing this exception for dir in s3fs.

Reply:
Okay, looks like the s3fs is also just throwing the FileNotFoundError when the dir is not existing. 
https://github.com/fsspec/s3fs/blob/main/s3fs/core.py#L948-L952